util/context.go

```diff
@@ -0,0 +1,43 @@
+package util
+
+import (
+	"context"
+
+	"github.com/grpc/grpc-go/metadata"
+
+	"github.com/infobloxopen/atlas-app-toolkit/auth"
+	"github.com/infobloxopen/atlas-app-toolkit/gateway"
+	"github.com/infobloxopen/atlas-app-toolkit/requestid"
+)
+
+// ForwardContext set to outgoing context request_id, auth_token and X-Forwarded-For header value
+func ForwardContext(ctx context.Context, token string) context.Context {
+	params := map[string][]string{}
+	if token != "" {
+		params[auth.AuthorizationHeader] = []string{token}
+	}
+
+	if rid, ok := requestid.FromContext(ctx); ok {
+		params[requestid.DefaultRequestIDKey] = []string{rid}
+	}
+
+	if rIPs, ok := gateway.Header(ctx, gateway.XForwardedFor); ok {
+		params[gateway.XForwardedFor] = []string{rIPs}
+	}
+
+	return ForwardContextWithCustomParams(ctx, params)
+}
+
+// ForwardContextWithCustomParams set to outgoing context all params from incoming context + custom params
+func ForwardContextWithCustomParams(ctx context.Context, params map[string][]string) context.Context {
```

Comment:
I'm not sure this function is needed. Usage of [google.golang.org/grpc/metadata](https://godoc.org/google.golang.org/grpc/metadata) is nearly as simple:
```go
md, _ := metadata.FromIncomingContext(ctx)
md = md.Join(params)
outgoingCtx := metadata.NewOutgoingContext(context.Background(), md)
```